plugins/webhooks/index.js

```diff
@@ -634,53 +646,71 @@ async function createEvents(eventFactory, userFactory, pipelineFactory,
     const events = [];
     const meta = createMeta(parsed);
 
-    for (let i = 0; i < pipelines.length; i += 1) {
-        const p = pipelines[i];
-        /* eslint-disable no-await-in-loop */
-        const pipelineBranch = await p.branch;
-        /* eslint-enable no-await-in-loop */
-        const startFrom = determineStartFrom(action, type, branch, pipelineBranch);
+    const pipelineTuples = await Promise.all(pipelines.map(async (p) => {
+        const tuple = { branch: await p.branch, pipeline: p };
+
+        return tuple;
+    }));
+
+    const ignoreExtraTriggeredPipelines = pipelineTuples.filter((tuple) => {
+        const startFrom = determineStartFrom(action, type, branch, tuple.branch);
```

Comment:
can you add this `startFrom` to the tupe so that you can avoid computing this again at https://github.com/screwdriver-cd/screwdriver/pull/1759/files#diff-4bff1c7e1da44fb8e755afbf06260240R670

Reply:
Thank you. I fixed.